xapian-applications/omega/assistant.cc

```diff
@@ -0,0 +1,127 @@
+/** @file assistant.cc
+ * @brief Worker module for putting text extraction into a separate process.
+ */
+/* Copyright (C) 2011 Olly Betts
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+#include <csignal>
+
+#include "worker_comms.h"
+#include "handler_pdf.h"
+#include "safeunistd.h"
+
+using namespace std;
+
+const int FD = 3;
+const int time_limit = 300;
+
+#if defined SIGALRM
+// IF I have SIGALRM (Best option)
+
+static void
+timeout_handler(int n) {
+    exit(n);
```

Comment:
It's not safe to use `exit()` in a signal handler (but `_exit()` or `_Exit()` are OK) - see `man 7 signal-safety` for a list.

`_Exit()` is the ISO C was (since C99) so probably the more portable choice:

https://en.cppreference.com/w/c/program/_Exit

Reply:
I didn't know that. I am going to chage it.